### PR TITLE
Add support for Python 3

### LINKED ISSUE #158: Add support for Python 3

### PR DESCRIPTION
**Pull request by pbauer**
Fixes #158

### CODE REVIEW

**Comment by pbauer** [comment]
@avoinea The tests are still partly failing but functionality is there.

**Comment by pbauer** [comment]
The change to use `publishTraverse` instead of `allowed_interface` in https://github.com/eea/eea.facetednavigation/pull/159/commits/be2ac04fc536cd7a888caa0d6735875a2a2cb87c is a terrible idea and a very temporary workaround until https://github.com/plone/Products.CMFPlone/issues/2624 is fixed or a better workaround is found.

**Comment by pbauer** [comment]
Some of the remaining test-failures are not stable. They might come from `FacetedVersion._digest` that seems to return different hashes during different test-runs in py3. That seems to mess up the cache. It might also be due to the dict `query` not being the same. 

Currently I have no more time to work on this. 

It also requires a test-setup with py2and some more manual testing.

File: eea/facetednavigation/widgets/tagscloud/counter.py [hunk 1]
```diff
@@ -38,17 +39,17 @@ def query(self, cid, **kwargs):
...
 
-        res = res.items()
+        res = list(res.items())
         res.sort(key=operator.itemgetter(1), reverse=True)
 
         maxitems = widget.maxitems
         if maxitems:
             res = res[:maxitems]
-        res.sort(key=operator.itemgetter(0), cmp=compare)
+        res.sort(key=lowercase)
```
Comment (pbauer) [inline eea/facetednavigation/widgets/tagscloud/counter.py]: This is probably wrong since lowercase uses the second element for the comparison.

**Comment by avoinea** [comment]
> It also requires a test-setup using tox with py2 and py3 and some more manual testing.

@pbauer Any good example?

**Comment by pbauer** [comment]
> > It also requires a test-setup using tox with py2 and py3 and some more manual testing.
> 
> @pbauer Any good example?

The is no canonical example yet. https://github.com/collective/collective.ifttt/pull/82 might be the right direction.

**Comment by pbauer** [comment]
@avoinea bobtemplates.plone now creates a test-setup that allows to test various plone and python-versions. One example with a complete setup is https://github.com/collective/collective.edtf_behavior

**Comment by jensens** [comment]
This branch is also needed for Plone 5.2 under Python 2.7. It addresses changes from Zope 2 -> 4 as well, needed to this addon!

**Comment by jensens** [comment]
Btw,. what is missing to merge this?

**Comment by avoinea** [comment]
Thank you all for this PR. I will take a look and merge if everything OK.

**Pull request merged**